### PR TITLE
[vcpkg-scripts] Update msys2

### PR DESCRIPTION
**Pull request by dg0yt**
Semi-automated update from `vcpkg-ci-msys2[*]`.

Needed to stay ahead of expiring downloads.

Major changes:
gfortran -> 13.2
msys2-runtime -> 3.4.9-3 (upstreamed fixes for flaky "Bad address" errors when invoking Windows program from MSYS2)
pkgconf -> 2.1 (upstreamed fixes for search order and link lib order)
python -> 3.11

### CODE REVIEW

**Comment by dg0yt** [comment]
Hm, the final ffmpeg error is 
~~~
D:/installed/x64-windows/tools/pkgconf/pkgconf.exe --exists --print-errors aom >= 1.0.0
/d/buildtrees/ffmpeg/src/n6.0-c7510d6860.clean/configure: line 995: D:/installed/x64-windows/tools/pkgconf/pkgconf.exe: Bad address
ERROR: aom >= 1.0.0 not found using pkg-config
~~~
but earlier invocations of `pkgconf` worked, and the `Bad adress` errors start earlier, too:
~~~
check_func_headers bzlib.h BZ2_bzlibVersion -lbz2
test_ld cc -lbz2
test_cc
BEGIN ./ffconf.IrXuJGdW/test.c
    1	#include <bzlib.h>
    2	#include <stdint.h>
    3	long check_BZ2_bzlibVersion(void) { return (long) BZ2_bzlibVersion; }
    4	int main(void) { int ret = 0;
    5	 ret |= ((intptr_t)check_BZ2_bzlibVersion) & 0xFFFF;
    6	return ret; }
END ./ffconf.IrXuJGdW/test.c
cl.exe -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_USE_MATH_DEFINES -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_WARNINGS -DPIC @D:/buildtrees/ffmpeg/x64-uwp-rel/cflags.rsp -nologo -DWINAPI_FAMILY=WINAPI_FAMILY_APP -D_WIN32_WINNT=0x0A00 -DHAVE_UNISTD_H=0 -ID:/installed/x64-uwp/include -c -Fo./ffconf.IrXuJGdW/test.o ./ffconf.IrXuJGdW/test.c
/d/buildtrees/ffmpeg/src/n6.0-c7510d6860.clean/configure: line 995: /c/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe: Bad address
~~~

The common thing is that both `pkgconf.exe` and `cl.exe` are invoked from within MSYS2 `bash`, so it could be an MSYS2 runtime issue.

**Comment by dg0yt** [comment]
> Non of the pkgconf PR went through so there seem to be still bugs in latest pkgconf.

Unrelated to "Bad address".
 - Upstream pkgconf has a search order regression since 2.0.0: https://github.com/pkgconf/pkgconf/issues/326, https://github.com/pkgconf/pkgconf/pull/327. 
   MSYS2 fixes this with a patch now. That's why I made this PR *now*.
- The remaining regressions in #34228 are from CMake reordering link libraries which are imported from pc files because it doesn't assume any dependencies. I fixed this for graphviz by adding dependencies unless they are explicit, https://github.com/microsoft/vcpkg/pull/34228/commits/6a15df6e2db83655487aeab69c30c89ded166e93.
  IIRC this started in some CMake 3.2x. This may explain why `qtinterfaceframework` has been flaky before. The relevant link libraries are picked via pkgconfig in qtmultimedia... IIRC there was a connection to link groups support, but I can't remember what's the chicken and what's the egg.
  I'm really close to forking or wrapping `FindPkgConfig.cmake` :-(

**Comment by dg0yt** [comment]
Nightly CI is now complaining about libcrypt being removed the msys2 index. Time to update.
But pkgconf 2.0 is still unusable because of https://github.com/pkgconf/pkgconf/issues/322. I will try to remove this part of the update.

File: scripts/cmake/vcpkg_acquire_msys.cmake [hunk 1]
```diff
@@ -426,31 +426,31 @@ macro(z_vcpkg_acquire_msys_declare_all_packages)
...
-        DEPS coreutils libcrypt sh
+        URL "https://mirror.msys2.org/msys/x86_64/perl-5.36.1-2-x86_64.pkg.tar.zst"
+        SHA512 0aead6f3df994c880726bec4c32b80f09b01ad0acaeb58f6a37d073f3278d7362eca70f6cdd1276a3838e9fc0b2f11f62e135a1139eef3f7b0ddd2520addff3b
+        DEPS coreutils libxcrypt sh
     )
     z_vcpkg_acquire_msys_declare_package(
-        URL "https://mirror.msys2.org/msys/x86_64/pkgconf-1.9.4-1-x86_64.pkg.tar.zst"
-        SHA512 0a5f0d69eec591a00d1aee985458dd855100184ec845b076d8f22ca51ba106964b8cf5b0061df288cdd611aa6a6e5fcb98eafded1c46536a0d17253240966f15
+        URL "https://mirror.msys2.org/msys/x86_64/pkgconf-2.0.2-2-x86_64.pkg.tar.zst"
+        SHA512 a2e79b647f003e7a807acedfd8f97a61e7f837a4ff2b543774d06b1403ff1dcf9b381c1b555d95207fd13441f01f4b6e486eea60a69dc207eb99fdc7983a6910
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_acquire_msys.cmake:453]:
```suggestion
        URL "https://mirror.msys2.org/msys/x86_64/pkgconf-1.9.4-1-x86_64.pkg.tar.zst"
        SHA512 0a5f0d69eec591a00d1aee985458dd855100184ec845b076d8f22ca51ba106964b8cf5b0061df288cdd611aa6a6e5fcb98eafded1c46536a0d17253240966f15
```

File: scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake [hunk 2]
```diff
@@ -22,16 +22,16 @@ elseif(CMAKE_HOST_WIN32)
             vcpkg_acquire_msys(PKGCONFIG_ROOT
                 NO_DEFAULT_PACKAGES
                 DIRECT_PACKAGES
-                    "https://mirror.msys2.org/mingw/clangarm64/mingw-w64-clang-aarch64-pkgconf-1~1.8.0-2-any.pkg.tar.zst"
-                    f682bbeb4588a169a26d3c9c1ce258c0022954fa11a64e05cd803bcbb8c4e2442022c0c6bc7e54d3324359c80ea67904187d4cb3b682914f5f14a03251daae7c
+                    "https://mirror.msys2.org/mingw/clangarm64/mingw-w64-clang-aarch64-pkgconf-1~2.0.3-2-any.pkg.tar.zst"
+                    7e9015f5a650706c75523255753b31f4ec32ead7a06b09adf5a611ad4cbb155756b234c70b461f9f28e5f9fe032269cbfd0bee4f011510c127d56474ed547f61
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:26]:
```suggestion
                    "https://mirror.msys2.org/mingw/clangarm64/mingw-w64-clang-aarch64-pkgconf-1~1.8.0-2-any.pkg.tar.zst"
                    f682bbeb4588a169a26d3c9c1ce258c0022954fa11a64e05cd803bcbb8c4e2442022c0c6bc7e54d3324359c80ea67904187d4cb3b682914f5f14a03251daae7c
```

File: scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake [hunk 3]
```diff
@@ -22,16 +22,16 @@ elseif(CMAKE_HOST_WIN32)
...
             )
             set("${program}" "${PKGCONFIG_ROOT}/clangarm64/bin/pkg-config.exe" CACHE INTERNAL "")
         else()
             vcpkg_acquire_msys(PKGCONFIG_ROOT
                 NO_DEFAULT_PACKAGES
                 DIRECT_PACKAGES
-                    "https://mirror.msys2.org/mingw/mingw32/mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst"
-                    e5217d9c55ede4c15706b4873761cc6e987eabc1308120a3e8406571ae2993907f3776f2b2bba18d7aaec80ef97227696058cedc1b67a773530dc1e6077b95e6
+                    "https://mirror.msys2.org/mingw/mingw32/mingw-w64-i686-pkgconf-1~2.0.3-2-any.pkg.tar.zst"
+                    637cfb4594cdabe09f419a58c2727e11c61930f2949d904bcc3dcd45de238d23e1c547e0b2b364bd405501014d1cc971cbfe006ace0991a0498eeca24af98c97
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:34]:
```suggestion
                    "https://mirror.msys2.org/mingw/mingw32/mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst"
                    637cfb4594cdabe09f419a58c2727e11c61930f2949d904bcc3dcd45de238d23e1c547e0b2b364bd405501014d1cc971cbfe006ace0991a0498eeca24af98c97
```

File: scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake [hunk 4]
```diff
@@ -31,7 +31,7 @@ elseif(CMAKE_HOST_WIN32)
                 NO_DEFAULT_PACKAGES
                 DIRECT_PACKAGES
                     "https://mirror.msys2.org/mingw/mingw32/mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst"
-                    e5217d9c55ede4c15706b4873761cc6e987eabc1308120a3e8406571ae2993907f3776f2b2bba18d7aaec80ef97227696058cedc1b67a773530dc1e6077b95e6
+                    637cfb4594cdabe09f419a58c2727e11c61930f2949d904bcc3dcd45de238d23e1c547e0b2b364bd405501014d1cc971cbfe006ace0991a0498eeca24af98c97
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:34]:
```suggestion
                    e5217d9c55ede4c15706b4873761cc6e987eabc1308120a3e8406571ae2993907f3776f2b2bba18d7aaec80ef97227696058cedc1b67a773530dc1e6077b95e6
```

**Comment by dg0yt** [comment]
My fixes for pkgconf and msys2-runtime are finally accepted into upstream sources... I hope the flaky "Bad Address" issues are resolved then.

**Comment by dg0yt** [comment]
jkqtplotter: baseline regression, https://github.com/microsoft/vcpkg/pull/35319

**Comment by MehdiChinoune** [comment]
Could you take the opportunity and use mingw-w64-i686-pkgconf/libwinpthreads on x86 and mingw-w64-x86_64-pkgconf on x64
somthing similar to https://github.com/microsoft/vcpkg/pull/24160, except that the file is scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake

**Comment by dg0yt** [comment]
> Could you take the opportunity and use mingw-w64-i686-pkgconf/libwinpthreads on x86 and mingw-w64-x86_64-pkgconf on x64 somthing similar to #24160, except that the file is scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake

`vcpkg_find_acquire_program(PKGCONFIG).cmake` is subject of #35299. 

These PRs used to have different blockers. That's why they exist in isolation. I can create a third, integrated PR. But I don't want to delay any of these isolated PRs now. Every day of delay means more users which will run into future msys2 download errors from not updating vcpkg early enough. And every day is another chance for the vcpkg maintainers to merge new baseline regressions...

**Comment by dg0yt** [comment]
Included in https://github.com/microsoft/vcpkg/pull/35331 (proposed).

**Comment by dg0yt** [comment]
Merged with #35331.

**Pull request closed without merging**